src/main/scala/com/softwaremill/realworld/db/Db.scala

```diff
@@ -0,0 +1,21 @@
+package com.softwaremill.realworld.db
+
+import com.zaxxer.hikari.{HikariConfig, HikariDataSource}
+import io.getquill.{SnakeCase, SqliteZioJdbcContext}
+import zio.ZLayer
+
+import javax.sql.DataSource
+
+object Db:
+
+  private def create(dbConfig: DbConfig): DataSource = {
+    val poolConfig = new HikariConfig()
+    poolConfig.setJdbcUrl(dbConfig.jdbcUrl)
+    new HikariDataSource(poolConfig)
+  }
+
+  // Used for migration and executing queries.
+  val dataSourceLive: ZLayer[DbConfig, Nothing, DataSource] = ZLayer.fromFunction(create(_))
+
+  // Quill framework object used for specifying sql queries.
+  def quillLive: ZLayer[Any, Nothing, SqliteZioJdbcContext[SnakeCase]] = ZLayer.succeed(new SqliteZioJdbcContext[SnakeCase](SnakeCase))
```

Comment:
this doesn't re-use the data source - isn't the connection pool created twice?

Reply:
Hmm I've looked in the logs and I see one datasource init there. I think quill doesn't use the datasource as its dependency, it uses it only during the execution -> https://github.com/softwaremill/realworld-tapir-zio/blob/0f380bfdc0ab2869d047c5560fcda3daf2079738/src/main/scala/com/softwaremill/realworld/articles/ArticlesRepository.scala#L14

Reply:
ah ... can we move `DataSource` to be a constructor-injected dependency, like all others?

Reply:
I've tried to find a way, but I failed. Quill seems to depend on the environment internally like f.e. here -> https://github.com/zio/zio-protoquill/blob/master/quill-jdbc-zio/src/main/scala/io/getquill/context/qzio/ZioJdbcContext.scala#L190
I don't see a way to externalize it.

Reply:
hm well you can always `.provide` the dependency to get an effect without deps

Reply:
Ok, I think I figured it out, it looks a little strange, but it seems to work -> https://github.com/softwaremill/realworld-tapir-zio/blob/9fd788272dad6571d95a28bc3b00669be2e300f6/src/main/scala/com/softwaremill/realworld/articles/ArticlesRepository.scala#L13

Reply:
Looks good, at least until Quill updates :)